### PR TITLE
Fix contract deployment issues

### PR DESCRIPTION
**Pull request by Mrtenz**

### CODE REVIEW

File: src/components/TransactionFlow/displays/FromToAccount.tsx [hunk 1]
```diff
@@ -94,7 +94,7 @@ const FromToAccount = ({ networkId, fromAccount, toAccount, displayToAddress = t
           />
         </AddressWrapper>
       </AddressContainer>
-      {displayToAddress && (
+      {displayToAddress && toAccount.address && (
```
Comment (FrederikBolding) [inline src/components/TransactionFlow/displays/FromToAccount.tsx:97]: This should be unnecesary https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/components/TransactionFlow/TxReceipt.tsx#L419

Perhaps the txType was missing causing the issue?

File: src/features/DeployContracts/stateFactory.tsx [hunk 2]
```diff
@@ -68,9 +68,10 @@ const DeployContractsFactory: TUseStateReducerFactory<DeployContractsState> = ({
       chainId: network.chainId,
       nonce
     });
+
     // check if transaction fails everytime
     await getGasEstimate(network, transaction);
-    transaction.gasLimit = gasLimit;
+    transaction.gasLimit = addHexPrefix(bigify(gasLimit).toString(16));
```
Comment (FrederikBolding) [inline src/features/DeployContracts/stateFactory.tsx:74]: Use `inputGasLimitToHex`

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: src/components/TransactionFlow/displays/FromToAccount.tsx (hunk 1 above)
Comment (Mrtenz) [reply src/components/TransactionFlow/displays/FromToAccount.tsx:97]: There is no `txReceipt` (which is used to get the `txType`) at all it looks like. I'm not sure what a good way to solve that would be?

Comment (FrederikBolding) [reply src/components/TransactionFlow/displays/FromToAccount.tsx:97]: This should be set here?
https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/features/DeployContracts/stateFactory.tsx#L99-L135

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**